addons/hr_holidays/models/hr_leave_allocation.py

```diff
@@ -384,25 +383,46 @@ def _get_accrual_plan_level_work_entry_prorata(self, level, start_period, start_
             work_entry_prorata = worked / (left + planned_worked) if (left + planned_worked) else 0
         return work_entry_prorata
 
-    def _process_accrual_plan_level(self, level, start_period, start_date, end_period, end_date):
-        """
-        Returns the added days for that level
-        """
-        self.ensure_one()
+    def _compute_added_value(self, level, start_period, start_date, end_period, end_date, calendar):
+
         if level.frequency == 'hourly' or level.accrual_plan_id.is_based_on_worked_time:
-            work_entry_prorata = self._get_accrual_plan_level_work_entry_prorata(level, start_period, start_date, end_period, end_date)
+            work_entry_prorata = self._get_accrual_plan_level_work_entry_prorata(level, calendar, start_period, start_date, end_period, end_date)
             added_value = work_entry_prorata * level.added_value
         else:
             added_value = level.added_value
+
         # Convert time in hours to time in days in case the level is encoded in hours
         if level.added_value_type == 'hour':
             added_value = added_value / (self.employee_id.sudo().resource_id.calendar_id.hours_per_day or HOURS_PER_DAY)
+
         period_prorata = 1
         if (start_period != start_date or end_period != end_date) and not level.accrual_plan_id.is_based_on_worked_time:
             period_days = (end_period - start_period)
             call_days = (end_date - start_date)
             period_prorata = min(1, call_days / period_days) if period_days else 1
-        return added_value * period_prorata
+
+        added_value = added_value * period_prorata
+        if level.frequency != 'hourly':
+            added_value *= (calendar.work_time_rate / 100)
+
+        return added_value
+
+    def _process_accrual_plan_level(self, level, start_period, start_date, end_period, end_date):
+        """
+        Returns the added days for that level
+        """
+        self.ensure_one()
+        datetime_min_time = datetime.min.time()
+        employee_timezone = timezone(self.employee_id.tz)
```

Comment:
on an other task we don't use employee timezone by default. https://github.com/odoo/odoo/pull/183093/files#diff-1c37517a76b393d1d30c2b03e96611643a747d12c13b086653989f4068c660b2
You can check the method _get_calendar_tz :)
So I don't know maybe ask Geoffrey :thinking: ; at least he will be aware of this difference :)